Desktop 4.4.2 release notes (#14093)
docker/docker.github.io: desktop/mac/release-notes/index.md

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### Security

> - Fixed CVE-2021-45449 that affects users currently on Docker Desktop version 4.3.0 or 4.3.1

Reviewer: Should we link the CVE here? Or maybe the security docs page? cc @jerae-duffin

Author: I added the link to the CVE as we haven’t yet added a section for it to our docs page.
Obviously we can change that later.

> - Fixed CVE-2021-45449 that affects users currently on Docker Desktop version 4.3.0 or 4.3.1

Reviewer 2: Can we add a link to the CVE?

Author: Yes, good point!

### New

> - Easy, Secure sign in with Auth0 and Single Sign-on

Reviewer 3: This is unclear IMHO.
`Single Sign-On on docker Hub`
will be more direct.

Reviewer 4: Perhaps we should link to the documentation for that; https://docs.docker.com/single-sign-on/ (I think)

Author: >  Single Sign-On on hub.docker.com

It’s not just SSO, the username/password flow has changed as well from the in-app dialog to a web page.

Author: Added the link to the docs!

>   - Single Sign-On (SSO) has been introduced for users with a Business Subscription.

Reviewer 2: ```suggestion
  - Single Sign-on: Users with a Docker Business subscription can now configure SSO to authenticate using their identity providers (IdPs) to access Docker. For more information, see [Single Sign-on](../../../single-sign-on/index.md).
```

Author: Awesome, thanks!

> chip](https://desktop.docker.com/mac/main/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){:
> .button .primary-btn }

### Security

- Fixed [CVE-2021-45449](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-45449) that affects users currently on Docker Desktop version 4.3.0 or 4.3.1.

> 

Reviewer 4: I'm wondering if for CVE's we should also include a summary of the vulnerability (and possible recommendations) in the changeling itself, so that users don't have to look elsewhere to collect this information.

From the CVE description:

```markdown
Docker Desktop version 4.3.0 and 4.3.1 contained has a bug that may log sensitive
information (access token or password) on the user's machine during login. This
only affects users if they are on Docker Desktop 4.3.0, 4.3.1 and the user has
logged in while on 4.3.0, 4.3.1. Gaining access to this data would require having
access to the user's local files.
```

And possibly recommendations; something like (please check/review - it's just a quick write-up);

```markdown
Docker Desktop 4.3.2 fixes this bug and remove old log files during upgrade.
While the sensitive information only leaked to files local to the user's machine,
we recommend users to verify if this data has not been stored in external location,
such as back-ups, and to rotate their Docker Hub password in case of concerns.
```

Author: Should we just copy the CVE description then?

Reviewer 2: @mat007 Discussed this with @stephanierifai and suggest that we add the following CVE description? 

"Docker Desktop version 4.3.0 and 4.3.1 has a bug that may log sensitive information (access token or password) on the user's machine during login. This only affects users if they are on Docker Desktop 4.3.0, 4.3.1 and the user has logged in while on 4.3.0, 4.3.1. Gaining access to this data would require having access to the user’s local files."

Author: > Could you also add this to the 4.3.2 release notes please?

Good catch!

Author: How do we want to format that? Do we still need the bullet list? If not where do we put the link?

Author: I’ve kept both, tell me what you think!

Reviewer 2: LGTM.Thanks

### New

- Easy, Secure sign in with Auth0 and Single Sign-on
  - Single Sign-on: Users with a Docker Business subscription can now configure SSO to authenticate using their identity providers (IdPs) to access Docker. For more information, see [Single Sign-on](../../../single-sign-on/index.md).
  - Signing in to Docker Desktop now takes you through the browser so that you get all the benefits of auto-filling from password managers.

### Upgrades

- [Docker Engine v20.10.12](https://docs.docker.com/engine/release-notes/#201012)
- [Compose v2.2.3](https://github.com/docker/compose/releases/tag/v2.2.3)
- [Kubernetes 1.22.5](https://github.com/kubernetes/kubernetes/releases/tag/v1.22.5)

### Bug fixes and minor changes

- It is now an error for `registry.json` to contain more than one organization in the `allowedOrgs` field. If you are using multiple organizations for different groups of developers, please provision a different `registry.json` file for each group.
- Fixed the memory statistics for containers in the Dashboard. Fixes [docker/for-mac/#4774](https://github.com/docker/for-mac/issues/6076).

> - Added a deprecated option to `settings.json`: `"deprecatedCgroupsv1": true`, which switches the Linux environment back to cgroups v1. This option will be removed in future. If you have software which still requires cgroups v1, please update it to be compatible with cgroups v2.

Reviewer 2: ```suggestion
- Added a deprecated option to `settings.json`: `"deprecatedCgroupsv1": true`, which switches the Linux environment back to cgroups v1. This option will be removed in future releases. If your software requires cgroups v1, you must update it to be compatible with cgroups v2.
```

- Fixed a regression in Compose that reverted the container name separator from `-` to `_`. Fixes [docker/compose-switch](https://github.com/docker/compose-switch/issues/24).

> - Fixed an issue where sleeping the machine after pausing Docker Desktop, it could not be resumed after waking up. Fixes [for-mac#6058](https://github.com/docker/for-mac/issues/6058).

Reviewer 2: ```suggestion
- Fixed an issue where putting the machine to Sleep mode after pausing Docker Desktop results in a failure to resume the machine from the Sleep mode. Fixes [for-mac#6058](https://github.com/docker/for-mac/issues/6058).
```

Author: The machine wakes up all right, but Docker cannot resume from pause.
What about:
```suggestion
- Fixed an issue where putting the machine to Sleep mode after pausing Docker Desktop results in Docker Desktop not being able to resume from pause after the machines comes out of Sleep mode. Fixes [for-mac#6058](https://github.com/docker/for-mac/issues/6058).
```
?

Reviewer 2: Sounds good, Mathieu.  Just one nit: after the **machine** comes out of Sleep...

Reviewer: LGTM
